.github/workflows/ci.yml

```diff
@@ -178,27 +143,36 @@ jobs:
         run: |
           ${{ inputs.caching_after_step_command}}
 
-  sast:
-    name: SAST Scan using semgrep
-    if: ${{ !contains( inputs.excluded_jobs, 'sast' ) && (github.actor != 'dependabot[bot]') }}
+
+
+  Security-scans:
+    name: Security scans
     runs-on: ubuntu-latest
     permissions: write-all
+    needs: caching
+    if: ${{!contains(github.actor, 'github-actions[bot]')  && (github.event_name != 'issue_comment') && (github.actor != 'dependabot[bot]') }}
```

Comment:
github-actions[bot] check is happening twice in this `if` statement itself. Should be checked only once.
And combine all the checks into contains() func itself.